fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -263,15 +263,9 @@ public void vectorizedAnalyze(Analyzer analyzer) {
 
     private boolean canCompareDate(PrimitiveType t1, PrimitiveType t2) {
         if (t1.isDateType()) {
-            if (t2.isDateType() || t2.isStringType()) {
-                return true;
-            }
-            return false;
+            return t2.isDateType() || t2.isStringType();
         } else if (t2.isDateType()) {
-            if (t1.isStringType()) {
-                return true;
-            }
-            return false;
+            return t1.isStringType();
```

Comment:
why not modify to  “return t1.isDateType() || t1.isStringType()”

Reply:
I will fix it.